From: Juergen Beisert <jbe@pengutronix.de>
Subject: Remove useless code if debug is not enabled

If one do not want to debug something, why still providing the code to do so?

Signed-off-by: Juergen Beisert <jbe@pengutronix.de>

---
 drivers/video/s3c2410fb.c |   11 ++++++++---
 1 file changed, 8 insertions(+), 3 deletions(-)

Index: linux-3.14/drivers/video/s3c2410fb.c
===================================================================
--- linux-3.14.orig/drivers/video/s3c2410fb.c
+++ linux-3.14/drivers/video/s3c2410fb.c
@@ -47,7 +47,7 @@
 #ifdef CONFIG_FB_S3C2410_DEBUG
 static int debug	= 1;
 #else
-static int debug;
+static  const int debug	= 0;
 #endif
 
 #define dprintk(msg...) \
@@ -588,6 +588,7 @@ static int s3c2410fb_blank(int blank_mod
 	return 0;
 }
 
+#ifdef CONFIG_FB_S3C2410_DEBUG
 static int s3c2410fb_debug_show(struct device *dev,
 				struct device_attribute *attr, char *buf)
 {
@@ -617,6 +618,7 @@ static int s3c2410fb_debug_store(struct 
 }
 
 static DEVICE_ATTR(debug, 0666, s3c2410fb_debug_show, s3c2410fb_debug_store);
+#endif
 
 static struct fb_ops s3c2410fb_ops = {
 	.owner		= THIS_MODULE,
@@ -981,11 +983,12 @@ static int s3c24xxfb_probe(struct platfo
 		goto free_cpufreq;
 	}
 
+#ifdef CONFIG_FB_S3C2410_DEBUG
 	/* create device files */
 	ret = device_create_file(&pdev->dev, &dev_attr_debug);
 	if (ret)
 		dev_err(&pdev->dev, "failed to add debug attribute\n");
-
+#endif
 	dev_info(&pdev->dev, "fb%d: %s frame buffer device\n",
 		fbinfo->node, fbinfo->fix.id);
 
@@ -1049,9 +1052,11 @@ static int s3c2410fb_remove(struct platf
 	iounmap(info->io);
 
 	release_mem_region(info->mem->start, resource_size(info->mem));
+#ifdef CONFIG_FB_S3C2410_DEBUG
 	device_remove_file(&pdev->dev, &dev_attr_debug);
-
+#endif
 	framebuffer_release(fbinfo);
+	platform_set_drvdata(pdev, NULL);
 
 	return 0;
 }
